src/game/FleeingMovementGenerator.cpp

```diff
@@ -47,7 +47,7 @@ void FleeingMovementGenerator<T>::_setTargetLocation(T& owner)
     PathFinder path(&owner);
     path.setPathLengthLimit(30.0f);
     path.calculate(x, y, z);
-    if (path.getPathType() & PATHFIND_NOPATH)
+    if (path.getPathType() & (PATHFIND_NOPATH | PATHFIND_SHORT))
```

Comment:
Ok, This is a bit too much, PATHFIND_SHORT is returned pretty often and in a large majority of case it dont cause any trouble. I think this change will have more bad side than good side. Remember even in retail there was some weird fleeing case ;)